### ISSUE TITLE
[FG:InPlacePodVerticalScaling] Container resize policy feature introduced in v1.27 doesn't interrupt CrashLoopBackOff

### ISSUE THREAD

**Issue by daqo98** [kubernetes/kubernetes#119838]
### What happened?

Hi, does someone know how to “break” the CrashLoopBackOff container status? I don’t want to wait all the backoff time nor delete the whole pod. I’m using the new features of restarting the container after resizing introduced in Kubernetes v1.27, but it seems that after I resize the container using the Python client, it still waits for the back-off time to finish before trying to run the container again.

Deployment info:

```spec:
      containers:
        - name: ...
          image: systemautoscaler/prime-numbers:dev
          imagePullPolicy: Always
          ports:
            - containerPort: ...
          resizePolicy:
            - resourceName: cpu
              restartPolicy: RestartContainer
            - resourceName: memory
              restartPolicy: RestartContainer
```

### What did you expect to happen?

No matter the state of the container, when I resize it, it should be restarted, even if waiting for Back Off.

### How can we reproduce it (as minimally and precisely as possible)?

1. Create a deployment with the spec field as specified above.
2. Scale down container resources to 1m CPU and to 1Mi memory (both limits and requests) using the Kubernetes API e.g. `kubernetes.client.CoreV1Api().patch_namespaced_pod(name=pod_name, namespace=namespace, body=new_pod_data)`
3. Wait till the container is in CrashLoopBackOff and then try to scale up the container resources with enough CPU and memory to run successfully.

### Kubernetes version

<details>

```console
$ kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.4", GitCommit:"fa3d7990104d7c1f16943a67f11b154b71f6a132", GitTreeState:"clean", BuildDate:"2023-07-19T12:20:54Z", GoVersion:"go1.20.6", Compiler:"gc", Platform:"windows/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.3", GitCommit:"25b4e43193bcda6c7328a6d147b1fb73a33f1598", GitTreeState:"clean", BuildDate:"2023-06-15T00:36:28Z", GoVersion:"go1.20.5", Compiler:"gc", Platform:"linux/amd64"}

```

</details>

### Cloud provider

<details>
Working locally on KinD cluster
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# BuildNumber  Caption                                    OSArchitecture  Version
19045        Microsoft Windows 10 Home Single Language  64 bits         10.0.19045
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by kannon92** [comment]
Not sure who worked on that feature but I see @vinaykul on the latest with this.

**Comment by mmiranda96** [comment]
/retitle Container resize policy feature introduced in v1.27 doesn't interrupt CrashLoopBackOff

**Comment by SergeyKanzhelev** [comment]
xref to the enhancement: #1287

**Comment by kannon92** [comment]
> xref to the enhancement: #1287

I had no idea how this was related to the issue.  I realize that you meant to include [#1287](https://github.com/kubernetes/enhancements/issues/1287) rather than k/k.

**Comment by SergeyKanzhelev** [comment]
Let me re-categorize this as this is a feature for the InPlaceUpdate enhancement

/retitle [FG:InPlacePodVerticalScaling] Container resize policy feature introduced in v1.27 doesn't interrupt CrashLoopBackOff
/remove-kind bug
/kind feature

**Comment by hshiina** [comment]
#125757 may fix this issue.

**Comment by esotsal** [comment]
@daqo98 please check screencast https://github.com/kubernetes/kubernetes/pull/125757#discussion_r1806712531 of partial fix. Please comment if you have time.

**Comment by esotsal** [comment]
Good news i think this will be fixed by #125757 @daqo98 ( as @hshiina predicted :-) )  please see related screencasts [here](https://github.com/kubernetes/kubernetes/pull/125757#issuecomment-2462698830)
